### PR TITLE
use cmp.Diff instead of reflect.DeepEqual

### PR DESCRIPTION
**Pull request by sivchari**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

[It](https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md) is stated that `Use cmp.Diff instead of reflect.Equal, to provide useful comparisons.`
So I replaced reflect.DeepEqual to cmp.Diff.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/extender_test.go [hunk 1]
```diff
@@ -345,9 +345,8 @@ func TestSchedulerWithExtenders(t *testing.T) {
 					t.Errorf("Unexpected error: %v", err)
 					return
 				}
-
-				if !reflect.DeepEqual(result, test.expectedResult) {
-					t.Errorf("Expected: %+v, Saw: %+v", test.expectedResult, result)
+				if diff := cmp.Diff(result, test.expectedResult); diff != "" {
```
Comment (AxeZhan) [inline pkg/scheduler/extender_test.go:348]:
```suggestion
				if diff := cmp.Diff(test.expectedResult,result); diff != "" {
```

File: pkg/scheduler/extender_test.go [hunk 2]
```diff
@@ -418,6 +417,7 @@ func TestIsInterested(t *testing.T) {
 }
 
 func TestConvertToMetaVictims(t *testing.T) {
+	t.Parallel()
```
Comment (AxeZhan) [inline pkg/scheduler/extender_test.go:420]: This seems irrelevant. Can we only focus on the replacement of `deepEqual` in this pr?

Comment (AxeZhan) [reply pkg/scheduler/extender_test.go:420]: Also, it seems that there are many changes of code style which results in a lot diff in git. 
Can we revert them as this pr is only focus on the replacement of deepEqual, may be we can do the style cleanup in another pr.

Comment (sivchari) [reply pkg/scheduler/extender_test.go]: You are right. I reverted it

File: pkg/scheduler/framework/plugins/volumebinding/binder_test.go [hunk 3]
```diff
@@ -2455,9 +2455,9 @@ func TestGetEligibleNodes(t *testing.T) {
 		eligibleNodes := testEnv.binder.GetEligibleNodes(logger, scenario.pvcs)
 
 		// Validate
-		if reflect.DeepEqual(scenario.eligibleNodes, eligibleNodes) {
```
Comment (AxeZhan) [inline pkg/scheduler/framework/plugins/volumebinding/binder_test.go:2458]: This seems incorrect? Why L2458 has the same content as L2462 ?

Comment (sivchari) [reply pkg/scheduler/framework/plugins/volumebinding/binder_test.go:2458]: Sorry, I've mistook.

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 4]
```diff
@@ -988,21 +985,24 @@ func TestPreFilterDisabled(t *testing.T) {
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
 	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterInterPodAffinity" from cycleState: %w`, framework.ErrNotFound))
-	if !reflect.DeepEqual(gotStatus, wantStatus) {
-		t.Errorf("status does not match: %v, want: %v", gotStatus, wantStatus)
+	if wantStatus.Code() != gotStatus.Code() {
```
Comment (AxeZhan) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: Compare of two status is always a problem to us.

Maybe this func can help instead of comparing each field in tests(Not sure).
```
// Equal checks equality of two statuses. This is useful for testing with
// cmp.Equal.
func (s *Status) Equal(x *Status) bool {
```

Comment (AxeZhan) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: Also, seems we defined some `cmpopts` for comparing of two status. 
```
var cmpOpts = []cmp.Option{
	cmp.Comparer(func(s1 *framework.Status, s2 *framework.Status) bool {
		if s1 == nil || s2 == nil {
			return s1.IsSuccess() && s2.IsSuccess()
		}
		return s1.Code() == s2.Code() && s1.Plugin() == s2.Plugin() && s1.Message() == s2.Message()
	}),
}
```
Maybe it's time to unify this behavior? 
WDYT @kerthcet @sanposhiho 
@sivchari Let's wait other reviewers to reach an agreement on this.

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: We can just use cmp.Diff, can't we?
`cmp.Diff` takes `Equal` function into consideration:
> Types with an Equal method (e.g., [time.Time.Equal](https://pkg.go.dev/time#Time.Equal)) may use that method to determine equality. This allows package authors to determine the equality operation for the types that they define.
https://pkg.go.dev/github.com/google/go-cmp/cmp

Comment (AxeZhan) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: Hmmm, from the doc it seems yes. However, when I try to remove `opts`, I got an error:
`FAIL: TestFilterPluginsWithNominatedPods/node_has_a_high-priority_nominated_pod_and_filters_fail `
Maybe it's something wrong in our `Equal` method🤦.

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: We should modify `Equal` method then, rather than having `opts` every places where we're using `cmp.Diff`.

Comment (AxeZhan) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: Yep, in `Equal` method, we compare errors of two status, which will not be true in most cases.
```
	if !cmp.Equal(s.err, x.err, cmpopts.EquateErrors()) {
		return false
	}
```
What's your suggestion? Should we just compare `err.ERROR()` here?

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: We should put the root error in test cases. 

Supposing we have this function, which we want to test:

```go
var ErrNotFound = errors.New("not found")

func Hoge() *framework.Status {
...
return framework.AsStatus(fmt.Errorf("hogehoge goes wrong: %w", ErrNotFound))
}
```

Then, in the test case, we can set `framework.AsStatus(ErrNotFound)` to `wantStatus`. 
I believe then `Equal` method works as it is.

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:988]: Some functions may not have root errors like the above `ErrNotFound` though, then we should modify the implementation to have root errors. That's the ideal state.
